MyPerfectOnboardingApplication/Sources/MyPerfectOnboarding.Services/Services/EditingService.cs

```
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading.Tasks;
+using MyPerfectOnboarding.Contracts.Models;
+using MyPerfectOnboarding.Contracts.Services.Generators;
+using MyPerfectOnboarding.Contracts.Services.ListItem;
+
+namespace MyPerfectOnboarding.Services.Services
+{
+    internal class EditingService : IEditingService
+    {
+        private readonly IListCache _cache;
+        private readonly ITimeGenerator _timeGenerator;
+
+        public EditingService(IListCache cache, ITimeGenerator timeGenerator)
+        {
+            _cache = cache;
+            _timeGenerator = timeGenerator;
+        }
+
+        public async Task ReplaceItemAsync(Guid id, ListItem editedItem)
+        {
+            var cachedItem = await _cache.GetItemAsync(id);
+
+            var updatedItem = UpdateItem(cachedItem, editedItem);
```

Before calling this method, you should check that your `_cache` returned a reasonable value. Always throw an exception "manually" rather than let `NullReferenceException` to be thrown – it allows you to give more detailed/contextual information on the reason/root cause of the situation that resulted in an unexpected state… (I expect you to tell me why won't you fix this one 😉)